src/realm/error_codes.cpp

```diff
@@ -187,7 +251,68 @@ static const std::map<std::string_view, ErrorCodes::Error> error_codes_map = {
     {"InvalidReadOnlySchemaChange", ErrorCodes::InvalidReadOnlySchemaChange},
     {"InvalidExternalSchemaChange", ErrorCodes::InvalidExternalSchemaChange},
     {"DeleteOnOpenRealm", ErrorCodes::DeleteOnOpenRealm},
-    {"NotSupported", ErrorCodes::NotSupported}};
+    {"NotSupported", ErrorCodes::NotSupported},
+    {"ClientUserNotFound", ErrorCodes::ClientUserNotFound},
+    {"ClientUserNotLogged_in", ErrorCodes::ClientUserNotLoggedIn},
+    {"ClientAppDeallocated", ErrorCodes::ClientAppDeallocated},
+    {"BadToken", ErrorCodes::BadToken},
+    {"MalformedJson", ErrorCodes::MalformedJson},
+    {"MissingJsonKey", ErrorCodes::MissingJsonKey},
+    {"BadBsonParse", ErrorCodes::BadBsonParse},
+
+    {"CustomError", ErrorCodes::CustomError},
+    {"MissingAuthReq", ErrorCodes::MissingAuthReq},
+    {"InvalidSession", ErrorCodes::InvalidSession},
+    {"UserAppDomainMismatch", ErrorCodes::UserAppDomainMismatch},
+    {"DomainNotAllowed", ErrorCodes::DomainNotAllowed},
+    {"ReadSizeLimitExceeded", ErrorCodes::ReadSizeLimitExceeded},
+    {"InvalidParameter", ErrorCodes::InvalidParameter},
+    {"MissingParameter", ErrorCodes::MissingParameter},
+    {"TwilioError", ErrorCodes::TwilioError},
+    {"GCMError", ErrorCodes::GCMError},
+    {"HTTPError", ErrorCodes::HTTPError},
+    {"AWSError", ErrorCodes::AWSError},
```

Comment:
Some of these error codes can be removed since we no longer support these external services.

Reply:
Maybe the server never sends these anymore, but it still seems to track these codes. I think the authoritative list is here: https://github.com/10gen/baas/blob/master/common/errors.go#L82. It might be easiest to just support handling all these on the client side rather than tracking each one down individually in the server code to check if it is sent or not.